src/Forms/GridFieldQueuedExportButton.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+namespace SilverStripe\GridfieldQueuedExport\Forms;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_ActionProvider;
+use SilverStripe\Forms\GridField\GridField_FormAction;
+use SilverStripe\Forms\GridField\GridField_HTMLProvider;
+use SilverStripe\Forms\GridField\GridField_URLHandler;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\Security\Security;
+use SilverStripe\View\ArrayData;
+use SilverStripe\View\Requirements;
+use Symbiote\QueuedJobs\DataObjects\QueuedJobDescriptor;
+use Symbiote\QueuedJobs\Services\QueuedJob;
+use Symbiote\QueuedJobs\Services\QueuedJobService;
+use SilverStripe\GridfieldQueuedExport\Jobs\GenerateCSVJob;
+
+/**
+ * A button you can add to a GridField to export that GridField as a CSV. Should work with any sized GridField,
+ * as the export is done using a queuedjob in the background.
+ */
+class GridFieldQueuedExportButton implements GridField_HTMLProvider, GridField_ActionProvider, GridField_URLHandler
+{
+
+    /**
+     * @var array Map of a property name on the exported objects, with values being the column title in the CSV file.
+     * Note that titles are only used when {@link $csvHasHeader} is set to TRUE.
+     */
+    protected $exportColumns;
+
+    /**
+     * @var string
+     */
+    protected $csvSeparator = ",";
+
+    /**
+     * @var boolean
+     */
+    protected $csvHasHeader = true;
+
+    /**
+     * Fragment to write the button to
+     */
+    protected $targetFragment;
+
+    /**
+     * @param string $targetFragment The HTML fragment to write the button into
+     * @param array $exportColumns The columns to include in the export
+     */
+    public function __construct($targetFragment = "after", $exportColumns = null)
+    {
+        $this->targetFragment = $targetFragment;
+        $this->exportColumns = $exportColumns;
+    }
+
+    /**
+     * Place the export button in a <p> tag below the field
+     */
+    public function getHTMLFragments($gridField)
+    {
+        $button = GridField_FormAction::create(
+            $gridField,
+            'export',
+            _t('SilverStripe\\Forms\\GridField\\GridField.CSVEXPORT', 'Export to CSV'),
```

Comment:
I think this could just be `__CLASS__ . 'CSVEXPORT'`, I don't see why keys need to span across modules when the thing they're identifying is specific to the one it's defined in... what do you think @robbieaverill ?

Reply:
If this is a valid translation from framework then it's good to use it rather than duplicating it. It's fine to use translations from other modules, provided they're dependencies of course!